node/src/accountant/test_utils.rs

```diff
@@ -21,10 +41,491 @@ pub fn make_receivable_account(n: u64, expected_delinquent: bool) -> ReceivableA
-        balance: (n * 1_000_000_000) as i64,
-        last_paid_timestamp: from_time_t(now - (n as i64)),
-        pending_payment_transaction: None,
+        wallet: recipient,
+        balance,
+        last_paid_timestamp: timestamp_opt.unwrap_or(SystemTime::now()),
+        pending_payment_rowid_opt: None,
+    }
+}
+
+//TODO should I replace this with the builder?
+pub fn make_accountant(
+    config_opt: Option<BootstrapperConfig>,
+    payable_dao_opt: Option<PayableDaoMock>,
+    receivable_dao_opt: Option<ReceivableDaoMock>,
+    pending_payments_dao_opt: Option<PendingPaymentsDaoMock>,
+    banned_dao_opt: Option<BannedDaoMock>,
+    persistent_config_opt: Option<PersistentConfigurationMock>,
+) -> Accountant {
+    let payable_dao_factory =
+        PayableDaoFactoryMock::new(Box::new(payable_dao_opt.unwrap_or(PayableDaoMock::new())));
+    let receivable_dao_factory =
+        ReceivableDaoFactoryMock::new(receivable_dao_opt.unwrap_or(ReceivableDaoMock::new()));
+    let pending_payments_dao_factory = PendingPaymentsDaoFactoryMock::new(
+        pending_payments_dao_opt.unwrap_or(PendingPaymentsDaoMock::default()),
+    );
+    let banned_dao_factory =
+        BannedDaoFactoryMock::new(banned_dao_opt.unwrap_or(BannedDaoMock::new()));
+    let mut subject = Accountant::new(
+        &config_opt.unwrap_or(BootstrapperConfig::new()),
+        Box::new(payable_dao_factory),
+        Box::new(receivable_dao_factory),
+        Box::new(pending_payments_dao_factory),
+        Box::new(banned_dao_factory),
+        Box::new(ConfigDaoFactoryMock::new(ConfigDaoMock::new())),
+    );
+    subject.persistent_configuration = if let Some(persistent_config) = persistent_config_opt {
+        Box::new(persistent_config)
+    } else {
+        Box::new(PersistentConfigurationMock::new())
+    };
+    subject
+}
+
+pub struct AccountantBuilder {
+    config: BootstrapperConfig,
+    payable_dao_factory: Box<dyn PayableDaoFactory>,
+    receivable_dao_factory: Box<dyn ReceivableDaoFactory>,
+    pending_payments_dao_factory: Box<dyn PendingPaymentsDaoFactory>,
+    banned_dao_factory: Box<dyn BannedDaoFactory>,
+    config_dao_factory: Box<dyn ConfigDaoFactory>,
+}
+
+impl Default for AccountantBuilder {
+    fn default() -> Self {
+        Self {
+            config: Default::default(),
+            payable_dao_factory: Box::new(PayableDaoFactoryMock::new(Box::new(
+                PayableDaoMock::new(),
+            ))),
+            receivable_dao_factory: Box::new(ReceivableDaoFactoryMock::new(
+                ReceivableDaoMock::new(),
+            )),
+            pending_payments_dao_factory: Box::new(PendingPaymentsDaoFactoryMock::new(
+                PendingPaymentsDaoMock::default(),
+            )),
+            banned_dao_factory: Box::new(BannedDaoFactoryMock::new(BannedDaoMock::new())),
+            config_dao_factory: Box::new(ConfigDaoFactoryMock::new(ConfigDaoMock::new())),
+        }
+    }
+}
+
+impl AccountantBuilder {
```

Comment:
Suggestion: Add methods that accept naked DAOs, rather than factories. Have those methods create factory mocks that return the supplied DAOs when instructed to make().

If these new methods make the old methods that accept factories obsolete, remove the old methods.